mypy/stubgen.py

```diff
@@ -866,17 +865,35 @@ def get_qualified_name(o: Expression) -> str:
 
 
 def walk_packages(packages: List[str]) -> Iterator[str]:
+    """Iterates through all packages and sub-packages in the given list.
+
+    Python packages have a __path__ attribute defined, which pkgutil uses to determine
+    the package hierarchy.  However, packages in C extensions do not have this attribute,
+    so we have to roll out our own.
+    """
     for package_name in packages:
         package = importlib.import_module(package_name)
         yield package.__name__
+        # get the path of the object (needed by pkgutil)
         path = getattr(package, '__path__', None)
         if path is None:
+            # object has no path; this means it's either a module inside a package
+            # (and thus no sub-packages), or it could be a C extension package.
+            if is_c_module(package):
+                # This is a C extension module, now get the list of all sub-packages
+                # using the inspect module
+                subpackages = [package.__name__ + "." + name
+                               for name, val in inspect.getmembers(package)
+                               if inspect.ismodule(val)]
```

Comment:
Hm, I just realized something: `inspect.getmembers()` only returns *current* members. With ordinary Python packages, at least, `import pkg` does not import all the submodules -- only those that are imported by the `__init__.py` or have been otherwise imported by other code running before. E.g. I ran this for the `http` module and it doesn't return anything.

Now this may not be relevant for `pybind11` or for your example package -- those seem to be comprised of all C extensions. Is that always the case? Or can a C extension module have a Python submodule? (Perhaps only if it also sets `__path__`, in which case we should be fine.)

Of course with the old way of doing things, no subpackages would be found either, so I guess it's still a net improvement.

Reply:
Good point, I did notice that importing a pybind11 package automatically import all the subpackages too, and was wondering if that's a pybind11 specific detail or it holds true for all C extensions in general.  I'll keep this in mind and see if there's a more robust solution.

Thanks for the quick response!